instrumentation/que/lib/opentelemetry/instrumentation/que/patches/poller.rb

```diff
@@ -0,0 +1,30 @@
+# frozen_string_literal: true
+
+# Copyright The OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+module OpenTelemetry
+  module Instrumentation
+    module Que
+      module Patches
+        # Instrumentation for the Que::Poller module
+        module Poller
+          def poll(*)
+            # Avoid tracing when should_poll? returns true. This is also used
+            # in Poller#poll to decide if the actuall poll should be executed
+            # or not. Without this we would generate a lot of unnecessary
+            # spans.
+            return unless should_poll?
+
+            if Que::Instrumentation.instance.config[:trace_poller]
+              Que::Instrumentation.instance.tracer.in_span('Que::Poller#poll') { super }
```

Comment:
In this case the code is is going to generate a span for `Que::Poller#poll` as well as the `pg` spans correct?

Is there a case where users would want to granular control and suppress the `pg` spans here as well?

Reply:
Correct. Good question. From my perspective, I don't see a need for a poll span without the pg span. The sidekiq instrumentation is following [the same pattern](https://github.com/open-telemetry/opentelemetry-ruby/blob/main/instrumentation/sidekiq/lib/opentelemetry/instrumentation/sidekiq/patches/poller.rb).